echopype/clean/impulse_noise.py

```diff
@@ -0,0 +1,463 @@
+
+    Returns
+    -------
+        bool: 2D array with IN mask
+        bool: 2D array with mask indicating valid IN mask samples.
+    """
+
+    # resample down vertically
+    iax_ = np.arange(iax[0], iax[-1], m)
+    Sv_ = mask_transformation.oned(Sv, iax, iax_, 0, log_var=True)[0]
+
+    # resample back to full resolution
+    jax = np.arange(len(Sv[0]))
+    Sv_, mask_ = mask_transformation.full(Sv_, iax_, jax, iax, jax)
+
+    # side comparison (±n)
+    dummy = np.zeros((iax.shape[0], n)) * np.nan
+    comparison_forward = Sv_ - np.c_[Sv_[:, n:], dummy]
+    comparison_backward = Sv_ - np.c_[dummy, Sv_[:, 0:-n]]
+
+    # get IN mask
+    comparison_forward[np.isnan(comparison_forward)] = np.inf
+    maskf = comparison_forward > thr
+    comparison_backward[np.isnan(comparison_backward)] = np.inf
+    maskb = comparison_backward > thr
+    mask = maskf & maskb
+
+    # get second mask indicating valid samples in IN mask
+    mask_[:, 0:n] = False
+    mask_[:, -n:] = False
+
+    return mask, mask_
+
+
+def _ryan_iterable(
+    Sv: Union[float, np.ndarray],
+    iax: Union[int, float, np.ndarray],
+    m: Union[int, float],
+    n: Tuple[int, ...],
+    thr: Union[int, float],
+) -> Tuple[np.ndarray, np.ndarray]:
+    """
+    Modified from "ryan" so that the parameter "n" can be provided multiple
+    times. It enables the algorithm to iterate and perform comparisons at
+    different n distances. Resulting masks at each iteration are combined in
+    a single mask. By setting multiple n distances the algorithm can detect
+    spikes adjacent each other.
+
+    Parameters
+    ----------
+        Sv  (float)     : 2D array with Sv data to be masked (dB).
+        iax (int, float): 1D array with i axis data (n samples or range).
+        m   (int, float): vertical binning length (n samples or range).
+        n   (int)       : number of pings either side for comparisons.
+        thr (int,float) : user-defined threshold value (dB).
+
+    Returns
+    -------
+        bool: 2D array with IN mask
+        bool: 2D array with mask indicating valid IN mask samples.
+    """
+
+    # resample down vertically
+    iax_ = np.arange(iax[0], iax[-1], m)
+    Sv_ = mask_transformation.oned(Sv, iax, iax_, 0, log_var=True)[0]
+
+    # resample back to full resolution
+    jax = np.arange(len(Sv[0]))
+    Sv_, mask_ = mask_transformation.full(Sv_, iax_, jax, iax, jax)
+
+    # perform side comparisons and combine masks in one unique mask
+    mask = np.zeros_like(Sv, dtype=bool)
+    for i in n:
+        dummy = np.zeros((iax.shape[0], i))
+        dummy[:] = np.nan
+        forward = Sv_ - np.c_[Sv_[:, i:], dummy]
+        backward = Sv_ - np.c_[dummy, Sv_[:, 0:-i]]
+        maskf = np.ma.masked_greater(forward, thr).mask
+        maskb = np.ma.masked_greater(backward, thr).mask
+        mask = mask | (maskf & maskb)
+
+    # get second mask indicating valid samples in IN mask
+    mask_[:, 0 : max(n)] = True
+    mask_[:, -max(n) :] = True
```

Comment:
Seems that this is a loop of computing `_ryan` masks? If so, how about reusing that function?

Reply:
Modified https://github.com/OceanStreamIO/echopype/commit/4a6d80fb0519e387be003b62e43d01ea07a25ba1